packages/internal/dex/sdk/src/exchange.ts

```diff
@@ -174,6 +280,18 @@ export class Exchange {
 
     const amountSpecified = newAmount(amount, tokenSpecified);
 
+    if (this.nativeTokenService.isWrapOrUnwrap(tokenIn, tokenOut)) {
+      // If the user is swapping between the native token and the wrapped native token,
+      // we want to just wrap/unwrap the native token instead of swapping
+      return this.getUnsignedWrapUnwrapTx(
```

Comment:
This will break the quote functionality. In that case, `if isWrap -> quoteToken = WIMX`. However, the case should be `if isWrap ^ tradeType = FROM_AMOUNT_OUT -> quoteToken = IMX`.

As in, if we wrap, but it's a "fromAmountOut" swap, the quote token should be the `tokenIn`. Specifying the `tokenIn` is the sufficient + necessary info for determining `isWrap`, and `otherToken` is the sufficient + necessary info for determining which token should be in the quote.

So we can still reduce by 1 parameter, and then the function looks like:
```
private async getUnsignedWrapUnwrapTx(
    tokenSpecified: Coin,
    amountIn: CoinAmount<Coin>,
    fromAddress: string,
    gasPrice: CoinAmount<Native> | null,
  ): Promise<TransactionResponse> {
    const isWrap = tokenSpecified.type === 'native';
    const quoteToken = isWrap ? this.wrappedNativeToken : this.nativeToken;
    const otherTokenCoinAmount = {
      token: quoteToken,
      value: amountIn.value,
    };
```

Thoughts?

Reply:
Ah yeah I get you, good point